### ISSUE TITLE
false positive error when callee has has an argument with a default value and **kwargs

### ISSUE THREAD

**Issue by micklat** [python/mypy#8052]
Bug: mypy reports type error on this code:

```
def foo(b: bool = False, **kwargs: str) -> None:
    print(b)

def bar(**kwargs: str):
    foo(**kwargs)

bar(user='joe')
```

The error reported is:
```
mypy_test.py:5: error: Argument 1 to "foo" has incompatible type "**Dict[str, str]"; expected "bool"
Found 1 error in 1 file (checked 1 source file)
```
I expect this file to type-check without errors.

mypy version: 0.750
python version: 3.7.3

I run mypy without flags.

**Comment by JelleZijlstra** [comment]
Mypy is right here, because a call to `bar(b="not a bool")` would result in `b` being of the wrong type at runtime.
